src/common/FlagsEnum.hpp

```diff
@@ -59,6 +59,13 @@ class FlagsEnum
         return static_cast<Q>(this->value_) & static_cast<Q>(flag);
     }
 
+    FlagsEnum &operator|(T flag)
```

Comment:
Why in the fuck does this `|` operator modify the object

Reply:
This pr should be marked as a draft - that's my bad. Didn't expect you to look into it